### ISSUE TITLE
Explore front-end framework options

### ISSUE THREAD

**Issue by MisterPhilip** [MisterPhilip/omnibug#118]
It'd be nice to potentially switch to using a front-end framework to speed up development of new features and allow easier contributions from others. Some potential options:

- [Vue](https://vuejs.org/)
- [React](https://reactjs.org/)
- [Preact](https://preactjs.com/)
- [Angular](https://angular.io/)

The build pipeline needs to be customizable enough to support:

- Multiple entry points: devtools panel, devtools page (largely empty wrapper, could do without a full build), and the options page
- Build single JS files, specifically `eventPage.js`
- Handle Typescript (see #117)
- Build out different manifest files

**Comment by johndavidsimmons** [comment]
+1 for something React based 🤑 . I will have to checkout preact

**Comment by MisterPhilip** [comment]
Yeah, Preact has (largely) the same API as React, but it's way smaller in size. We'd just need to do [some work](https://github.com/preactjs/preact-cli/issues/439#issuecomment-365061139) around the build pipeline and customizing it since they do not support multiple endpoints OOTB. And I need to learn React... but that should be easy enough!
